tests/test_schema.py

```diff
@@ -3,9 +3,10 @@
 from nomad.client import normalize_all, parse
 
 
-def test_schema():
+def test_schema(capture_from_logger, clean_up):
```

Comment:
This is what is confusing me a bit on using `capture_from_logger` as a fixture.

Do you want to capture if there is an ERROR in the logger? If so, I would suggest that: the functionality which is being tested should return something that we expect when there is an error in the logger.

Let me put an example. Imagine we define:
```python
def sum_two_positive_int(int1: int, int2: int, logger: BoundLogger) -> Optional[int]:
    if int1 < 0 or int2 < 0:
        logger.error('One of the integers is negative. Returning None.')
        return None
    return int1 + int2
```

Then testing should check:
```python
@pytest.mark.parametrize(
    'int1, int2, result',
    [
        (1, 1, 2),
        (-1, 1, None),
        (1, -1, None),
    ],
)
def test(self, int1, int2, result):
    """
    Test the `chemical_symbol` and `atomic_number` resolution for the `AtomsState` section.
    """
    assert sum_two_positive_int(int1, int2) == result
```

Thing is, if the logger stores an Error but we don't return after it, it will either crash later on the function because of some missing information or continue doing stuff that we don't want to (in the example before, the return will be 0, but maybe we don't want to run this functionality if some conditions are not satisfied).

Reply:
In that case, I will argue that there's missing a error handling before (tho in this specific case, it was a problem of the nomad source code, and hence you had to patch it there):
```python
if not hasattr(archive.m_context, 'NO_SUCH_FUNCTION'):
    logger.error('Could not find the function `NO_SUCH_FUNCTION()` defined in archive.m_context.')
    return None
```

Reply:
Hence the return... but yes, both raise / logger.error + return are equivalent.